test/Interop/Cxx/operators/member-inline.swift

```diff
@@ -90,4 +90,55 @@ OperatorsTestSuite.test("DifferentTypesArray.subscript (inline)") {
   expectEqual(1.5.rounded(.up), resultDouble.rounded(.up))
 }
 
+OperatorsTestSuite.test("ReadOnlyIntArrayByVal.subscript (inline)") {
+  var arr = ReadOnlyIntArrayByVal(1)
+
+  let result0 = arr[0]
+  let result2 = arr[2]
+  let result4 = arr[4]
+
+  expectEqual(1, result0)
+  expectEqual(3, result2)
+  expectEqual(5, result4)
+
+  arr.setValueAtIndex(42, 2)
+  let result5 = arr[2]
+  expectEqual(42, result5)
+}
+
+OperatorsTestSuite.test("DifferentTypesArrayByVal.subscript (inline)") {
+  var arr = DifferentTypesArrayByVal()
+
+  let resultInt: Int32 = arr[2]
+  let resultDouble: Double = arr[0.1]
+
+  expectEqual(3, resultInt)
+  expectEqual(1.5.rounded(.down), resultDouble.rounded(.down))
+  expectEqual(1.5.rounded(.up), resultDouble.rounded(.up))
+}
+
+OperatorsTestSuite.test("ReadOnlyNonTrivialArrayByVal.subscript (inline)") {
+  var arr = ReadOnlyNonTrivialArrayByVal()
+  let NonTrivialByVal = arr[0];
+  if let cStr = NonTrivialByVal.Str {
```

Comment:
I think it makes sense to force unwrap here as well. In tests, I think it makes sense to force unwrap a lot because you're basically asserting that something is there that you expect to be.